packages/core/custom-branding/core-custom-branding-server-internal/custom_branding_service.ts

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+import type { CustomBrandingFetchFn, CustomBrandingStart } from '@kbn/core-custom-branding-server';
+import type { CustomBranding } from '@kbn/core-custom-branding-common';
+import { KibanaRequest } from '@kbn/core-http-server';
+import type { CoreContext } from '@kbn/core-base-server-internal';
+import type { Logger } from '@kbn/logging';
+
+/**
+ * @internal
+ */
+export interface InternalCustomBrandingSetup {
+  register: (pluginName: string, fetchFn: CustomBrandingFetchFn) => void;
+  getBrandingFor: (request: KibanaRequest) => Promise<CustomBranding>;
+}
+
+export class CustomBrandingService {
+  private pluginName: string | undefined;
+  private logger: Logger;
+  private fetchFn: CustomBrandingFetchFn | undefined;
```

Comment:
NIT: The team usually prefers `private prop?: Foo` to `private prop: Foo | undefined`